gateway-sp-comms/src/single_sp.rs

```diff
@@ -99,6 +100,15 @@ const DISCOVERY_INTERVAL_IDLE: Duration = Duration::from_secs(60);
 // will require an MGS update.
 const TLV_RPC_TOTAL_ITEMS_DOS_LIMIT: u32 = 1024;
 
+// We allow our client to specify the max RPC attempts and the
+// per-attempt timeout; however, it's very easy to set a timeout that is
+// too low for the "reset the SP" request, especially if the SP being
+// reset is a sidecar (which means it won't be able to respond until it
+// brings the management network back online). We will override the max
+// attempt count for only that message to ensure we give SPs ample time
+// to reset.
+const SP_RESET_TIME_ALLOWED: Duration = Duration::from_secs(30);
```

Comment:
In the interest of not making things worse and always accidentally resetting to the old version, maybe we should make this even longer. What about 60, or even 120 seconds?

It only makes sense to extend the timeout if we keep trying periodically and don't have to wait the whole timeout though.

Reply:
This timeout didn't actually change – I just pulled it into a global constant so that I could use it in more than one place.